### PR TITLE
Plane: Quadplane: add Q_OPTION to ignore forward flight angle limits …

### PR DESCRIPTION
**Pull request by IamPete1**
…in Q modes

Currently Qmodes at constrained to the forward flight angle limits (except tailsitters), this adds a option to turn that off.

https://youtu.be/lUTjDNbkSqM

Funded by ARACE

### CODE REVIEW

File: ArduPlane/mode_qstabilize.cpp [hunk 1]
```diff
@@ -15,29 +15,30 @@ bool ModeQStabilize::_enter()
 void ModeQStabilize::update()
 {
     // set nav_roll and nav_pitch using sticks
-    int16_t roll_limit = MIN(plane.roll_limit_cd, plane.quadplane.aparm.angle_max);
+    float roll_input = plane.channel_roll->norm_input();
```
Comment (peterbarker) [inline ArduPlane/mode_qstabilize.cpp:18]:
```suggestion
    const float roll_input = plane.channel_roll->norm_input();
```

Making it clear values are fixed can simplify thinking about the rest of the logic....

**Comment by IamPete1** [comment]
Refactored to make more readable (hopefully)

File: ArduPlane/mode_qstabilize.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/mode_qstabilize.cpp]: done

**Comment by IamPete1** [comment]
rebased to restart the replay test, also tested all three configs, tailsitter and quaplanes with and without the new bit.

**Comment by IamPete1** [comment]
Retested in realight, is now all working, including taiksitter roll

**Review by tridge — APPROVED** [review]

**Pull request merged**